docs/src/capi/explanation/in-place-upgrades.md

```diff
@@ -0,0 +1,236 @@
+The main annotations that drive the upgrade process are as follows (for a 
+complete and up-to-date list of these annotations and their values please 
+refer to [annotations reference page][4]):
+
+- `v1beta2.k8sd.io/in-place-upgrade-to` : Instructs the controller to 
+perform an upgrade with the specified option/method. This is the only 
+annotation that we as the users need to put on the objects.
+- `v1beta2.k8sd.io/in-place-upgrade-status`: As soon as the controller 
+starts the upgrade process, the object will be marked with this 
+annotation to indicate the status of the upgrade. It can either be 
+`in-progress`, `failed` or `done`.
+- `v1beta2.k8sd.io/in-place-upgrade-release`: When the upgrade is 
+performed successfully, this annotation will indicate the current 
+Kubernetes release/version installed on the machine.
+
+From now on, let's use the following abbreviations to make this article 
+shorter and more readable:
+
+- `v1beta2.k8sd.io/in-place-upgrade-to` --> `upgrade-to`
+- `v1beta2.k8sd.io/in-place-upgrade-status` --> `status`
+- `v1beta2.k8sd.io/in-place-upgrade-release` --> `release`
+- `v1beta2.k8sd.io/in-place-upgrade-last-failed-attempt-at` --> 
+`last-failed-attempt-at`
+
+### Single Machine In-Place Upgrade Controller
+
+The Machine objects can be marked with the `upgrade-to` annotation to 
+trigger an in-place upgrade for that machine. As an example, let’s say 
+we have a machine with `1.30/stable` Kubernetes snap installed on it. 
+To initiate an in-place upgrade for this machine, we can annotate it 
+with `upgrade-to: channel=1.31/stable`. The single machine upgrader 
+(which is watching for changes on machines), notices this annotation 
+and attempts to upgrade the Kubernetes version of that machine to the 
+specified version.
+
+Because {{product}} is shipped in a snap package, performing an upgrade 
+can be as easy as doing a `snap refresh`. Upgrade methods or options 
+can be specified to upgrade to a snap channel, revision, or install a 
+new snap from a file already placed on the machine to make up for 
+air-gapped environments.
+
+When the upgrade is finished successfully, we will notice (at least) the 
+following annotations on the machine:
+
+```yaml
+annotations:
+  v1beta2.k8sd.io/in-place-upgrade-release: "channel=1.31/stable"
+  v1beta2.k8sd.io/in-place-upgrade-status: "done"
+```
+
+If the upgrade fails, the controller will mark the machine with the 
+following annotations and retry immediately:
+
+```yaml
+annotations:
+  # the `upgrade-to` causes the retry to happen
+  v1beta2.k8sd.io/in-place-upgrade-to: "channel=1.31/stable"
+  v1beta2.k8sd.io/in-place-upgrade-status: "failed"
+
+  # orchestrator will notice this annotation and knows that the 
+  # upgrade for this machine failed
+  v1beta2.k8sd.io/in-place-upgrade-last-failed-attempt-at: "Sat, 7 Nov 
+  2024 13:30:00 +0400"
+```
+
+For the detailed implementation of this controller, make sure to 
+checkout the [{{product}} CAPI repository][capi-repo].
```

Comment:
I would remove this, if anything add a link in the beginning where you mention the controller.